packages/flame/lib/src/widgets/sprite_widget.dart

```diff
@@ -89,13 +103,31 @@ class _SpriteWidgetState extends State<SpriteWidget> {
     final oldValue = await oldFutureValue;
     final newValue = await newFutureValue;
 
+    if (widget.rasterize && oldValue.image != newValue.image) {
+      oldValue.image.dispose();
```

Comment:
is there a condition here where rasterize was changed from true to false and then we fail to dispose here?

Reply:
oh, yeas, good catch, we need to check on the old value as well

Reply:
@luanpotter put a commit to address that, can you take another look please?